experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/proxies.spec.ts

```diff
@@ -277,5 +277,79 @@ describe("SharedTreeList", () => {
 				assert.deepEqual(listB, ["b0", "a0", "b1"]);
 			});
 		});
+
+		describe("between lists with overlapping types", () => {
+			const _ = new SchemaBuilder({
+				scope: "test",
+			});
+
+			const listA = _.list([_.string, _.number]);
+			const listB = _.list([_.number, _.boolean]);
+
+			const objectSchema = _.object("parent", {
+				listA,
+				listB,
+			});
+
+			const schema = _.intoSchema(objectSchema);
+
+			const initialTree = {
+				listA: ["a", 1],
+				listB: [2, true],
+			};
+
+			function getEitherList(
+				root: ProxyRoot<typeof schema>,
+				list: "a" | "b",
+			): ProxyNode<typeof listA> | ProxyNode<typeof listB> {
+				return list === "a" ? root.listA : root.listB;
+			}
```

Comment:
What value does this function provide? It looks like calls to `getEitherList(root, "a")` could just be replaced with `root.listA`, but I'm guessing the return type is needed for the operations to be allowed. If so it may be less confusing to just cast.

Reply:
Good question. It's checking that (A | B).move(A | B) compiles, rather than checking only that A.move(B) and B.move(A) compile. Currently, it doesn't make a difference, but if we decide to try and add stronger type checking (like we had before this change), it will catch compile errors that otherwise wouldn't be caught. This PR is motivated specifically by those errors that are caused by the union.